core/include/scipp/core/multi_index.h

```diff
@@ -30,276 +47,332 @@ auto get_slice_dim(const T &param, const Ts &... params) {
   return param ? param.dim : get_slice_dim(params...);
 }
 
-template <scipp::index N, size_t... I, class... StridesArgs>
-void copy_strides(std::array<std::array<scipp::index, NDIM_MAX>, N> &dest,
-                  const scipp::index ndim, std::index_sequence<I...>,
-                  const StridesArgs &... strides) {
-  (std::reverse_copy(strides.begin(), strides.begin() + ndim, dest[I].begin()),
-   ...);
-  // The last stride must be 0 for the end-state detection.
-  if (ndim < NDIM_MAX) {
-    ((dest[I][ndim] = 0), ...);
+template <scipp::index N_Operands>
+constexpr auto get_strides_buffer_size(const scipp::index ndim) noexcept {
+  return std::max(ndim, scipp::index{2}) * N_Operands;
+}
+
+constexpr auto get_shape_buffer_size(const scipp::index ndim) noexcept {
+  return std::max(ndim, scipp::index{2});
+}
+
+constexpr auto get_coord_buffer_size(const scipp::index ndim) noexcept {
+  return std::max(ndim, scipp::index{2});
+}
+
+template <scipp::index N_Operands>
+constexpr auto get_buffer_size(const scipp::index ndim) noexcept {
+  return get_strides_buffer_size<N_Operands>(ndim) +
+         get_shape_buffer_size(ndim) + get_coord_buffer_size(ndim);
+}
```

Comment:
No need for having these in the header, if only used in constructor?

Reply:
The constructor has a template, so it needs to see those functions. But if I switch back to statically allocated arrays, those functions become redundant anyway.